xline/src/storage/index.rs

```diff
@@ -11,41 +12,38 @@ use crate::server::command::{KeyRange, RangeType};
 #[derive(Debug)]
 pub(crate) struct Index {
     /// Inner struct of `Index`
-    inner: Mutex<IndexInner>,
+    inner: RwLock<IndexInner>,
```

Comment:
we could consider using a skiplist to achieve lockfree concurrent access https://docs.rs/crossbeam-skiplist/latest/crossbeam_skiplist/

Reply:
Ok, I'll do some researches and do it in another pr